### PR TITLE
fix: checkpoint export with custom python objects

### PR DESCRIPTION
**Pull request by sidneyw**
## Description
Re-apply https://github.com/determined-ai/determined/pull/459.

Fixes `tf.keras` trial checkpoint export by loading the users trial class, calling build_model, and loading the weights. This is necessary because custom object in keras are not serialized via `h5`. Loss functions and custom layers were therefore not recoverable in the previous iteration.

Custom layers in for torch trials were also not pickled correctly. Therefore, we are applying the same instantiate and load weights strategy to pytorch going forward.

[DET-3044]

## Test Plan
The current test suite covers these changes.

[DET-3044]: https://determinedai.atlassian.net/browse/DET-3044

### CODE REVIEW

**Comment by sidneyw** [comment]
moved to my fork https://github.com/determined-ai/determined/pull/548
